features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -898,6 +914,11 @@ nsapi_error_t AT_CellularNetwork::detach()
     _at.resp_start();
     _at.resp_stop();
 
+    _at.cmd_start("AT+COPS=2");
+    _at.cmd_stop();
+    _at.resp_start();
+    _at.resp_stop();
+
     call_network_cb(NSAPI_STATUS_DISCONNECTED);
```

Comment:
not in line with disconnect methods where we call NSAPI_STATUS_DISCONNECTED only in case of success